### PR TITLE
feat(connections): added return routing support for connections

### PR DESCRIPTION
**Pull request by tplooker**
#### Short description of what this resolves:

This PR adds support for establishing connections with return routing.

#### Changes proposed in this pull request:

- Added decoding invitiations from a valid URL format
- Added return routing support for connection establishment
- Updated AgentFramework to 4.0.0-preview

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added decoding invitiations from a valid URL format
+- Added return routing support for connection establishment
+- Updated AgentFramework to 4.0.0-preview
```
Comment (joshhill) [inline CHANGELOG.md]: In `### Updated` section and include the preview build no?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added decoding invitiations from a valid URL format
```
Comment (joshhill) [inline CHANGELOG.md]: Remove `Added` as its implied by section

File: src/Osma.Mobile.App.Services/AgentContext.cs [hunk 3]
```diff
@@ -33,5 +29,8 @@ public class AgentContext : IAgentContext
         /// </summary>
         /// <value>The agent verkey.</value>
         public string Verkey { get; set; }
+
+        /// <inheritdoc />
+        public IList<MessageType> SupportedMessages { get; set; } = new List<MessageType>();
```
Comment (joshhill) [inline src/Osma.Mobile.App.Services/AgentContext.cs:34]: SupportedMessageTypes

File: CHANGELOG.md (hunk 1 above)
Comment (tplooker) [reply CHANGELOG.md]: Because it is unreleased perhaps it is better to omit this from the changelog until we have a version for an update to occur from?

Comment (joshhill) [reply CHANGELOG.md]: Yeah, agreed

**Pull request merged**